lib/src/platform_client.dart

```diff
@@ -186,13 +191,52 @@ class PlatformClient {
       'context': jsonEncode(context),
       'requestSource': _config.clientId,
       'requestType': 'AIRA', // Required but unused.
+      'hasMessage': (null != message && message.isNotEmpty) || fileNames.isNotEmpty || true == cannotTalk,
+      'message': '$message${fileNames.isEmpty ? '' : ' (With files: $fileNames)'}',
+      'fileIds': fileIds,
+      'cannotTalk': cannotTalk ?? false,
       'useWebrtcRoom': true,
     });
 
     ServiceRequest serviceRequest =
         ServiceRequest.fromJson(await _httpPost('/api/user/$_userId/service-request', body));
 
-    return KurentoRoom.create(_config.environment, this, _session!, _pubnub, serviceRequest, roomHandler);
+    messagingClient?.serviceRequestId = serviceRequest.id;
+
+    return KurentoRoom.create(_config.environment, this, _session!, messagingClient, serviceRequest, roomHandler);
+  }
+
+  Future<List<String>> _sendPreCallMessage(String? text, Map<String, List<int>>? fileMap) async {
+    if (null == messagingClient) {
+      throw UnsupportedError('The application does not support messaging');
+    }
+    _log.finest('Sending pre-call message (message: $text, files: ${fileMap?.keys.join(', ')})');
+    String? message = text?.trim();
+    await messagingClient!.sendStart();
+    if (null != fileMap && fileMap.isNotEmpty) {
+      if (fileMap.length == 1) {
+        // If we have only one file, send it with the file.
+        var fileEntry = fileMap.entries.first;
+        SentFileInfo fileInfo = await messagingClient!.sendFile(fileEntry.key, fileEntry.value, text: message);
+        return [fileInfo.id];
+      } else {
+        // if we have multiple files, send them separately from teh message
```

Comment:
https://dart.dev/guides/language/effective-dart/documentation#do-format-comments-like-sentences. Also, "the" is misspelled.